File: Practical_Task_1/Wrong_Ones/InterestCalculator.java
public class interestCalculator { // Incorrect naming convention, should be InterestCalculator
    public double calculateInterest(double principal, int rate, int time) {
        double interest = 0;
        
        // Bug: Incorrect handling of negative values and zero time/rate
        if (time > 0 && rate > 0) {
            interest = (principal * rate * time) / 100;
        } else if (time < 0 || rate < 0) { // Confusing logic, does not handle all cases correctly
            System.out.println("Error: This code has bugs. Time or rate cannot be negative.");
            return -1;
        } else {
            System.out.println("Error: Rate or time is zero, interest calculation incorrect.");
        }
        
        // Inconsistent and unclear logic
        if (interest == 0) {
            System.out.println("Error: Interest calculation may be incorrect.");
        }
        
        return interest;
    }

    public static void main(String[] args) {
        InterestCalculator calculator = new InterestCalculator(); // Wrong class name instantiation
        double interest = calculator.calculateInterest(1000, 0, 2); // Test with zero rate
        System.out.println("Calculated Interest: " + interest);
    }
}

File: Practical_Task_1/Wrong_Ones/interestCalculator.php
<?php

class InterestCalculator {
    public function calculateInterest($principal, $rate, $time) {
        // Bug 1: Arithmetic error caused by division by zero (runtime error)
        if ($rate == 0 || $time == 0) {
            throw new ArithmeticError("Division by zero: Rate or time cannot be zero.\n");
        }

        // Bug 2: Throwing an uncaught exception for negative principal (exception handling error)
        if ($principal < 0) {
            throw new InvalidArgumentException("Principal cannot be negative.\n");
        }

        // Bug 3: Incorrect logic causing negative interest (logical error)
        $interest = ($principal * $rate * $time) / -100;  // This will return negative interest
        return $interest;
    }
}

// Handling the instantiation and testing the calculateInterest function
try {
    $calculator = new InterestCalculator();
    // Testing with a zero rate to trigger the ArithmeticError
    $interest = $calculator->calculateInterest(1000, 0, 2);
    echo "Calculated Interest: " . $interest . "\n";
} catch (ArithmeticError $e) {
    echo "Fatal ArithmeticError caught: " . $e->getMessage() . "\n";
    echo "Stack trace:\n" . $e->getTraceAsString() . "\n";
} catch (InvalidArgumentException $e) {
    echo "InvalidArgumentException caught: " . $e->getMessage() . "\n";
} catch (Exception $e) {
    echo "General Exception caught: " . $e->getMessage() . "\n";
}

?>

